app/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java

```diff
@@ -498,6 +501,25 @@ public boolean onPreferenceClick(Preference preference) {
                             .setNegativeButton(android.R.string.no, null)
                             .show();
                     return true;
+                case KEY_UNDO_IGNORED_DEVICES_FOLDERS:
+                    new AlertDialog.Builder(getActivity())
+                            .setMessage(R.string.undo_ignored_devices_folders_question)
+                            .setPositiveButton(android.R.string.yes, (dialog, which) -> {
+                                if (mApi == null) {
+                                    Toast.makeText(getActivity(),
+                                            getString(R.string.generic_error) + getString(R.string.syncthing_disabled_title),
+                                            Toast.LENGTH_SHORT).show();
+                                    return;
+                                }
+                                mApi.undoIgnoredDevicesAndFolders();
+                                mRequireRestart = true;
```

Comment:
Why needs a restart?

Reply:
It's like all the other options where "mRequireRestart = true": If this flag is true, the config is sent to syncthing on pressing home or leaving the SettingsActivity. This is our new settings apply routine we've introduced in PR #1099 . I rather wouldn't like to write extra code and try out if a just-send-the-config thing works. The user can pretty wait for a clean restart as this is rarely clicked.

Reply:
Btw undoIgnoredDevicesAndFolders only alters the in-memory cached config held by the wrapper not syncthing core. So the config must be sent by PostConfigRequest (see #1099) later which the flag indicates.

Reply:
We should rename the flag, its confusing.

Reply:
So I'll make a follow-up PR and rename it to "mPendingConfig" as that's what it indicates the config still needs to be sent to syncthing.

Reply:
You can just do it in this one

Reply:
Ok, here we go :) 34748e476d31d9bd8f874d87b70a98b6385d258f